tests/test_proto_log_exporter.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import inspect
+import logging
+import typing
+import unittest
+
+from opentelemetry.proto.logs.v1.logs_pb2 import (
+    LogsData,
+    SEVERITY_NUMBER_WARN,
+    SEVERITY_NUMBER_ERROR,
+    SEVERITY_NUMBER_FATAL,
+)
+from snowflake.telemetry._internal.exporter.otlp.proto.log_exporter import (
+    LogWriter,
+    LogWriterResult,
+    SnowflakeLoggingHandler,
+)
+
+
+class InMemoryLogWriter(LogWriter):
+    """Implementation of :class:`.LogWriter` that stores protobufs in memory.
+
+    This class is intended for testing purposes. It stores the deserialized
+    protobuf messages in a list in memory that can be retrieved using the
+    :func:`.get_finished_protos` method.
+    """
+
+    def __init__(self):
+        self._protos = []
+
+    def write_logs(self, serialized_logs: bytes) -> LogWriterResult:
+        message = LogsData()
+        message.ParseFromString(serialized_logs)
+        self._protos.append(message)
+        return LogWriterResult.SUCCESS
+
+    def get_finished_protos(self) -> typing.Tuple[LogsData, ...]:
+        return tuple(self._protos)
+
+    def clear(self):
+        self._protos.clear()
+
+
+class TestSnowflakeLoggingHandler(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
```

Comment:
With the stable API, the code in opentelemetry_logging.py will primarily be the first three lines of this method.